onnxruntime/core/graph/graph_flatbuffers_utils.h

```diff
@@ -93,20 +125,24 @@ Status LoadAttributeOrtFormat(const fbs::Attribute& fbs_attr,
 /// @param[in] ort_tensor ORT tensor to serialize to a flatbuffer tensor
 /// @param[in] builder flatbuffer builder to use for creating the flatbuffer tensor
 /// @param[out] fbs_tensor flatbuffer tensor to serialize the ORT tensor to
+/// @param[out] external_data_writer Optional delegate to write the tensor data to an external file
 /// @return Status indicating success or providing error information
 Status SaveOrtTensorOrtFormat(
     const std::string& tensor_name, const onnxruntime::Tensor& ort_tensor,
     flatbuffers::FlatBufferBuilder& builder,
-    flatbuffers::Offset<fbs::Tensor>& fbs_tensor);
+    flatbuffers::Offset<fbs::Tensor>& fbs_tensor,
+    ExternalDataWriter external_data_writer = nullptr);
 
 /// @brief Load an ORT tensor from a flatbuffer tensor
 /// @param[in] fbs_tensor flatbuffer tensor to load the ORT tensor from
 /// @param[in] allocator Allocator to use for creating the ORT tensor
 /// @param[out] tensor_name Name of the tensor
 /// @param[out] ort_tensor ORT tensor to load the flatbuffer tensor into
+/// @param[in] external_data_reader Optional delegate to read from an external data file
 /// @return Status indicating success or providing error information
 Status LoadOrtTensorOrtFormat(const fbs::Tensor& fbs_tensor, const AllocatorPtr allocator,
-                              std::string& tensor_name, onnxruntime::Tensor& ort_tensor);
+                              std::string& tensor_name, onnxruntime::Tensor& ort_tensor,
+                              const ExternalDataReader& external_data_reader = nullptr);
```

Comment:
A nullable reference does seem odd. References are usually for things that must be present.
A prototype above passes it by value. If it is an std::function, then it can be checked if it is present.
At the very least we want to make them consistent.

Reply:
Marking as resolved so that the PR can go in before the code freeze, but will note for future follow-up PR